tests/transforms/test_array_utils.py

```diff
@@ -132,3 +132,105 @@ def test_crop_array_raises_error(arr):
+                    [0, 0, 0, 0, 0, 0, 0],
+                    [0, 0, 0, 0, 0, 0, 0]]),
+                (5, 5),
+                ((1, 6, 1, 6), ((0, 0), (0, 0)))),
+            (
+                np.array([
+                    [0, 0, 0, 0, 0, 0, 0],
+                    [0, 0, 0, 0, 0, 0, 0],
+                    [0, 0, 1, 1, 0, 0, 0],
+                    [0, 0, 1, 1, 0, 0, 0],
+                    [0, 0, 0, 0, 0, 0, 0],
+                    [0, 0, 0, 0, 0, 0, 0],
+                    [0, 0, 0, 0, 0, 0, 0]]),
+                (5, 5),
+                ((1, 6, 1, 6), ((0, 0), (0, 0)))),
+            (
+                np.array([
+                    [0, 0, 0, 0, 0, 0, 0],
+                    [0, 0, 0, 0, 0, 0, 0],
+                    [0, 0, 0, 0, 0, 0, 0],
+                    [0, 0, 0, 1, 1, 0, 0],
+                    [0, 0, 0, 1, 1, 0, 0],
+                    [0, 0, 0, 0, 0, 0, 0],
+                    [0, 0, 0, 0, 0, 0, 0]]),
+                (5, 5),
+                ((2, 7, 2, 7), ((0, 0), (0, 0)))),
+            (
+                np.array([
+                    [1, 1, 0, 0, 0, 0, 0],
+                    [0, 0, 0, 0, 0, 0, 0],
+                    [0, 0, 0, 0, 0, 0, 0],
+                    [0, 0, 0, 0, 0, 0, 0],
+                    [0, 0, 0, 0, 0, 0, 0],
+                    [0, 0, 0, 0, 0, 0, 0],
+                    [0, 0, 0, 0, 0, 0, 0]]),
+                (5, 5),
+                ((0, 3, 0, 4), ((2, 0), (1, 0)))),
+            (
+                np.array([
+                    [0, 0, 0, 0, 0, 0, 0],
+                    [1, 1, 1, 1, 1, 1, 1],
+                    [1, 1, 1, 1, 1, 1, 1],
+                    [1, 1, 1, 1, 1, 1, 1],
+                    [1, 1, 1, 1, 1, 1, 1],
+                    [1, 1, 1, 1, 1, 1, 1],
+                    [0, 0, 0, 0, 0, 0, 0]]),
+                (2, 2),
+                ((3, 5, 3, 5), ((0, 0), (0, 0)))),
+            (
+                np.array([
+                    [0, 0, 0, 0, 0, 0, 0],
+                    [1, 1, 1, 1, 1, 1, 1],
+                    [1, 1, 1, 1, 1, 1, 1],
+                    [1, 1, 1, 1, 1, 1, 1],
+                    [1, 1, 1, 1, 1, 1, 1],
+                    [1, 1, 1, 1, 1, 1, 1],
+                    [0, 0, 0, 0, 0, 0, 0]]),
+                (3, 3),
+                ((2, 5, 2, 5), ((0, 0), (0, 0)))),
+            ])
+def test_content_extents(arr, shape, expected):
+    bounds = au.content_extents(arr, shape)
+    assert np.all(bounds == expected)
+
+
+@pytest.mark.parametrize(
+        ("arr", "shape"),
+        [
+            (
+                np.array([
+                    [0, 0, 0, 0, 0, 0, 0],
+                    [0, 0, 0, 0, 0, 0, 0],
+                    [0, 0, 1, 1, 1, 0, 0],
+                    [0, 0, 1, 2, 2, 0, 0],
+                    [0, 0, 1, 1, 3, 0, 0],
+                    [0, 0, 0, 0, 0, 0, 0],
+                    [0, 0, 0, 0, 0, 0, 0]]),
+                (5, 5)),
+            ])
+def test_compare_crop_pad_and_extents(arr, shape):
```

Comment:
I think this would be cleaner providing the expected array as a parameter rather than creating the indexed array in the test function

Reply:
the point of this test is that the two methods agree. That's what the comment is saying.